s2spy/traintest.py

```diff
@@ -1,105 +1,115 @@
 """s2spy train/test splitting methods.
 
-A collection of train/test splitting approaches for cross-validation.
+Wrapper around sklearn splitters for working with (multiple) xarray dataarrays.
 """
+from typing import Iterable
+from typing import List
 from typing import Optional
+from typing import Tuple
 from typing import Type
 from typing import Union
 import numpy as np
-import pandas as pd
 import xarray as xr
 from sklearn.model_selection._split import BaseCrossValidator
 
 
-def split_groups(
-    splitter: Type[BaseCrossValidator],
-    data: Union[xr.Dataset, pd.DataFrame],
-    key: Optional[str] = "anchor_year",
-):
-    """Splits calendar resampled data into train/test groups, based on the input key.
-    As splitter, a Splitter Class such as sklearn's KFold can be passed.
+# Mypy type aliases
+X = Union[xr.DataArray, List[xr.DataArray]]
```

Comment:
An interesting finding, `pylint` complains on my machine about single letter variable:
`Variable name "X" doesn't conform to snake_case naming stylepylint(invalid-name)`
But since it follows the vocabulary of machine learning and it doesn't cause confusion, I think that's why in sklearn people also does this. 

But just to be friendly to the users (especially from climate science community) who is new to the ML field, maybe we can add one line in the docstrings at the top: <br>
"In this module, X(or x) refers to training data and y refers to testing data. This is the convention widely used by the machine learning community and for more information please check the machine learning vocabulary of scikit-learn in their code examples: https://scikit-learn.org/stable/tutorial/basic/tutorial.html"

Feel free to add it if you think it is nice to mention.

Reply:
These Pylint warnings do not show up in prospector because [strictness](https://prospector.landscape.io/en/master/profiles.html#strictness-1) is set to medium. Setting it to high will trigger them in prospector as well.

After our discussion (we agreed to allow certain short/capitalized) variable names, I looked into options for configuring pylint. Interestingly, if I add configuration options in pyproject.toml, they are picked up by pylint run through prospector, but not by pylint run as standalone. Vice versa, adding them to setup.cfg yields the opposite result. By adding .pylintrc, both tools picked up the right configuration. 

@geek-yang I think vscode should also pick up the new rule. Can you verify?

Reply:
Thanks for the info. Lessons learned 😉 . Just checked vscode and yes, it picks up the new rule. Thanks for exploring this.